rfcs/002-baskets-specification.md

```diff
@@ -0,0 +1,137 @@
+
+#### Credit Class & Batch Hierarchy (Fig. 1)
+
+![](media/002-fig-1.jpg)
+
+# The Baskets Use Case
+
+When bringing ecocredits to market (e.g. offering them for sale on a decentralized exchange or to a potential buyer), there are a few issues we need to overcome:
+- A buyer may want to purchase more ecocredits than is issued in any one batch
+- A buyer may want to buy a large number of ecocredits without choosing which specific credit batches they come from
+- An ecocredit owner may want to place their ecocredits in a liquidity pool, which require a very high volume of a credits to be locked in liquidity (higher volume than available in any one batch)
+
+Baskets are intended to solve these problems, by aggregating heterogeneous ecocredits together. Baskets enable ecocredits from different batches to be deposited in a basket in exchange for redeemable vouchers, known as “basket tokens”. Any basket token is fully fungible with any other basket token from the same basket. And every basket token is redeemable for an underlying ecocredit that was originally put in the given basket.
+
+These basket tokens in aggregate hold the same value as the sum of all the underlying ecocredits in the basket, and the basket tokens can be brought to market in a more unified form for a more streamlined credit buyer experience, and for using ecocredits in high volume liquidity pools.
+
+Our hope is that these basket assets increase marketability of ecocredits in the following ways:
+- Basket criteria provide market signals by curating acceptable standards
+- Basket assets enable easy retail and corporate purchases at larger volumes
+- Conversely, basket assets enable easy selling at larger volumes for brokers, project developers, and holders
+- Basket assets enable greater composability in cryptocurrency/blockchain such as in Defi for use as collateral
+- Basket assets are more easily Interoperable with existing carbon token standards that are emerging in other cryptocurrency/blockchain projects (e.g. KlimaDAO, MOSS Token)
+
+# Specification
+
+### Basket Criteria & Instantiation
+
+Ecocredit baskets are containers/bundles for ecocredits meeting a specific set of criteria such as credit class, credit type, credit’s project location, credit’s vintage, etc. Baskets are instantiated with the following parameters:
+- Basket Name & Display Name
+- Basket Criteria - a set of attributes an ecocredit must have in order to be added to the basket
+- Exponent (used for converting decimal ecocredits to integer bank denom)
+- _disable_auto_retire_ - boolean that determines if credits are auto retired when removed from the basket
+- _enable_picking_ - boolean that enables or disables PickFromBasket functionality
```

Comment:
might be nice to describe what picking means here rather than just referencing `PickFromBasket` functionality